homeassistant/components/hydrawise/config_flow.py

```diff
@@ -22,25 +23,61 @@ class HydrawiseConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Hydrawise."""
 
     VERSION = 1
-
-    async def _create_entry(
-        self, api_key: str, *, on_failure: Callable[[str], ConfigFlowResult]
+    MINOR_VERSION = 2
+
+    def __init__(self) -> None:
+        """Construct a ConfigFlow."""
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def _create_or_update_entry(
+        self,
+        api_key: str | None = None,
+        username: str | None = None,
+        password: str | None = None,
+        *,
+        on_failure: Callable[[str], ConfigFlowResult],
     ) -> ConfigFlowResult:
         """Create the config entry."""
-        api = legacy.LegacyHydrawiseAsync(api_key)
+
+        # Verify that the provided credentials work."""
+        if api_key:
+            api = legacy.LegacyHydrawiseAsync(api_key)
+        else:
+            api = client.Hydrawise(auth.Auth(username, password))
         try:
             # Skip fetching zones to save on metered API calls.
-            user = await api.get_user(fetch_zones=False)
+            user = await api.get_user()
+        except NotAuthorizedError:
+            return on_failure("invalid_auth")
         except TimeoutError:
             return on_failure("timeout_connect")
         except ClientError as ex:
             LOGGER.error("Unable to connect to Hydrawise cloud service: %s", ex)
             return on_failure("cannot_connect")
 
         await self.async_set_unique_id(f"hydrawise-{user.customer_id}")
-        self._abort_if_unique_id_configured()
 
-        return self.async_create_entry(title="Hydrawise", data={CONF_API_KEY: api_key})
+        if not self.reauth_entry:
+            self._abort_if_unique_id_configured()
+
+            # We are creating an entry for username/password even if we
+            # import a legacy YAML file. This will require users to immediately
+            # re-authenticate.
```

Comment:
I think the comment is a bit misleading. We're creating an entry *without* username and password, right?

Reply:
Rewrote the comment.